infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -345,7 +351,17 @@ public void onClose(KubernetesClientException cause) {
    *     </ul>
    *     otherwise, it must be explicitly closed
    */
-  public CompletableFuture<Void> waitRunningAsync(String name) {
+  public CompletableFuture<Void> waitRunningAsync(String name, @Nullable Tracer tracer) {
+    final Span tracingSpan =
+        tracer == null
+            ? null
+            : tracer
+                .buildSpan("WaitRunningAsync")
```

Comment:
Could we move this code into a separate method? This would simplify the method

Reply:
Moved to `TracerUtil`